poetry/utils/appdirs.py

```diff
@@ -47,23 +75,20 @@ def user_cache_dir(appname):
 
         # Add our app name and Cache directory to it
         path = os.path.join(path, appname, "Cache")
-    elif sys.platform == "darwin":
-        # Get the base path
-        path = expanduser("~/Library/Caches")
-
-        # Add our app name to it
-        path = os.path.join(path, appname)
     else:
         # Get the base path
-        path = os.getenv("XDG_CACHE_HOME", expanduser("~/.cache"))
+        path = os.getenv(
+            "XDG_CACHE_HOME",
+            expanduser("~/Library/Caches" if sys.platform == "darwin" else "~/.cache"),
+        )
 
         # Add our app name to it
         path = os.path.join(path, appname)
 
     return path
 
 
-def user_data_dir(appname, roaming=False):
+def user_data_dir(appname="", roaming=False):
```

Comment:
What's this about?

Reply:
this was left over from some tests I did. Will post an update in a minute. The rational behind it, was this:

I might want to get just the `user_data_dir` directory, without any app appended to it, and the empty string would achieve that.

Reply:
And I just realized that `None` would do the same.